libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -281,8 +281,10 @@ bool HTTPClient::begin(String url)
 }
 #endif // HTTPCLIENT_1_1_COMPATIBLE
 
-bool HTTPClient::beginInternal(String url, const char* expectedProtocol)
+bool HTTPClient::beginInternal(const String& __url, const char* expectedProtocol)
 {
+    String url(__url);
```

Comment:
I assume this is to mimic the mutability nature of the method body? i.e.: it modifies the url argument.

Reply:
Yes, there is a bit of code later in this function trying to split the url apart which No longer works when it's not passed on by value 

There might be a way to avoid the extra copy but I wanted to do that as a second step in order to not pollute this review for that.